ast/term.go

```diff
@@ -2929,6 +2956,46 @@ func unmarshalExpr(expr *Expr, v map[string]interface{}) error {
 			expr.With = ws
 		}
 	}
+	if loc, ok := v["location"].(map[string]interface{}); ok {
+		expr.Location = &Location{}
+		if err := unmarshalLocation(expr.Location, loc); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func unmarshalLocation(loc *Location, v map[string]interface{}) error {
```

Comment:
I have added this since I needed to expand the functionality of term unmarshalling now that a location might have been included in the output data. I'm not sure if this is correct, perhaps it should not be possible to unmarshal and get the location data back in.

Reply:
If the only thing that this could affect is where error locations are reported for example, then I don't think it'll be an issue to leave in. If anyone else sees an issue though, I'll defer to them on this.